### ISSUE TITLE
Missing events when moving multiple folders 

### ISSUE THREAD

**Issue by SpringsTea** [gorakhargosh/watchdog#596]
Windows 10
Python 3.7.4
watchdog 0.9.0

Here is the minimum code I'm running:

```
import time
from watchdog.observers import Observer
from watchdog.events import FileSystemEventHandler

PATH='watch/'

class MyHandler(FileSystemEventHandler):
    def on_created(self, event):
        print(f'event type: {event.event_type}  path : {event.src_path} isdir : {event.is_directory}')
       

event_handler = MyHandler()
observer = Observer()
observer.schedule(event_handler, path=PATH, recursive=True)
observer.start()

try:
    while True:
        time.sleep(1)
except KeyboardInterrupt:
    observer.stop()
observer.join()
```
I am moving multiple folders with images into the `watch/` folder and getting unexpected results.
I move 3 folders, each contains 10 - 15 jpg images, each image is 4-5mb.
The result I get is one event trigger for each file in the FIRST folder (first as in alphabetically first, lets call it folder0) and then no subsiquent events trigger for the other folders.
I'm just copying images from one location on the local drive to another (copying, not moving) using the windows explorer.

Now here's where it gets kind of weird. I set `recursive=False`
Do the above use case, everything gets logged, no problem.

Next use case, in reality, some of these folder are going to have many more images. This time I add a folder with about 1000 images of similar size along side the previous 3 folders. In this case, about 73 files get logged from the new folder, and then that's it. The files will continue to transfer but no new events will trigger. It's not always 73, sometime it's more, some times it's lets, I'm assuming it has to do with `WATCHDOG_TRAVERSE_MOVED_DIR_DELAY`, and that is is stopping if the folder takes longer than 1 second to transfer. 

**Last use case**, I turn `recursive=True` back on again and move in the single 1000 file folder. Each file event gets logged. Great. So neither option can fully meet my use case. What is going on here? What am I to do?

TLDR;
`recursive=True` fails to log events for multiple folders, and `recursive=False` fails to log events for large directories

**Comment by axfelix** [comment]
I've also noticed some inconsistency on Windows lately with `recursive=True` when moving or deleting nested folders -- been thinking I could try to work around this in my own codebase, but it'd get messy fast, not sure if there's anything you can do upstream or if you're just capturing all of the kernel events you get from Windows already.

**Comment by jesuinovieira** [comment]
Same here. Any news?

**Comment by BoboTiG** [comment]
Nobody is working on it, but PR are very welcome.

To ease reproduction and digging, maybe a simple python script to create the tree and running watchdog to highlight misses would be a good step moving forward.

**Comment by BoboTiG** [comment]
Also I would say to try the latest release, to see if there were improvement for that use case.

**Comment by BoboTiG** [comment]
The situation may be fixed with the work done on #709. So you will have to use watchdog V1.0.0 or newer. PLease reopen if the issue persists.
